Scripts/prepare_release.py

```diff
@@ -0,0 +1,363 @@
+
+def git(cmd: str, *args: str, **kwargs) -> subprocess.CompletedProcess:
+    return subprocess.run(["git", cmd, *args], # pylint: disable=subprocess-run-check
+        **{"capture_output": True, "check": False, "encoding": "utf-8", **kwargs})
+
+A = TypeVar(name="A")
+def with_error(fn: Callable[[], A]) -> A:
+    try:
+        return fn()
+    except Exception as e:
+        progress(f"failed: {e}")
+        if isinstance(e, subprocess.CalledProcessError):
+            progress("Process output: ")
+            progress(e.stdout)
+            progress(e.stderr)
+        raise CannotReleaseError() from e
+
+def assert_one(msg: str, condition: Callable[[], Any]):
+    assert msg.endswith("?")
+    progress(msg + " ", end="")
+    if not SKIP_ALL_CHECKS and not with_error(condition):
+        progress("no, exiting.")
+        raise CannotReleaseError()
+    progress("yes.")
+
+def run_one(msg: str, operation: Callable[[], None]):
+    assert msg.endswith("...")
+    progress(msg + " ", end="")
+    with_error(operation)
+    progress("done.")
+
+def unwrap(opt: Optional[A]) -> A:
+    assert opt
+    return opt
+
+class NewsFragments:
+    IGNORED = {".gitignore", "README.md"}
+    KNOWN_EXTENSIONS = {".feat": "New features",
+                        ".fix": "Bug fixes"}
+
+    def __init__(self, path: str) -> None:
+        self.path = Path(path)
+        self.fragments: Dict[str, List[Path]] = {}
+        self.unrecognized: List[Path] = []
+        self.collect()
+
+    def collect(self) -> None:
+        if self.path.exists():
+            for f in self.path.iterdir():
+                if f.suffix in self.KNOWN_EXTENSIONS:
+                    self.fragments.setdefault(f.suffix, []).append(f)
+                elif f.name not in self.IGNORED:
+                    self.unrecognized.append(f)
+
+    def check(self):
+        if self.unrecognized:
+            names = ', '.join(quote(f.name) for f in sorted(self.unrecognized))
+            raise ValueError(f"Not sure what to do with {names}.")
+
+    def render(self) -> str:
+        rendered = []
+        for ext, title in self.KNOWN_EXTENSIONS.items():
+            if ext not in self.fragments:
+                continue
+            rendered.append(f"## {title}")
+            for pth in self.fragments[ext]:
+                text = pth.read_text(encoding="utf-8")
+                link = f"(https://github.com/dafny-lang/dafny/pull/{pth.name})"
+                entry = indent(f"- {text}\n{link}", "  ").lstrip()
+                rendered.append(entry)
+        return "\n\n".join(rendered)
+
+    def delete(self):
+        for _, paths in self.fragments.items():
+            for pth in paths:
+                pth.unlink()
+
+class Version(NamedTuple):
+    VERNUM_RE = re.compile("^(?P<prefix>[0-9]+[.][0-9]+[.][0-9]+)(?P<suffix>-.+)?$")
+
+    prefix: str # Main version number (1.2.3)
+    rdate: date # Release date
+    suffix: str # Optional marker ("alpha")
```

Comment:
I would prefer very much to have complete and explicit names, so that it does not feel like a compiled code :-)

```suggestion
    version_number: str # Main version number (1.2.3)
    release_date: date # Release date
    version_number_suffix: str # Optional marker ("alpha")
```

Reply:
In the same answer spirit, it's not clear with "prefix" or "suffix" what of the version numbers are included in which :-)

Maybe the following could reach a consensus?
`prefix` => `major_minor_patch`
`suffix` stays the same.

Reply:
OK, I renamed `prefix` to `main` and `suffix` to `identifier` (the name used for this in semantic versioning)